### PR TITLE
Password Recovery

### PR DESCRIPTION
**Pull request by RJFryman**
Recovery when given email function with tests, recovery when username given work in progress.

rails g devise:views
kept views/devise/mailer/reset_password, password/edit, passwords/new
added recoverable to app/models/user
modified config/initializer/devise to allow config.reset.keys_password to accept username as well as email.

### CODE REVIEW

File: spec/features/user_password_recovery_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  scenario "wrong" do
+    visit '/'
+    click_link "Sign in"
+    click_link "Forgot your password?"
+    fill_in "user_email", with: "not_real@example.com"
+    click_button "Send me reset password instructions"
+    page.should have_content "not found"
+  end
+end
+
```
Comment (elizamarcum) [inline spec/features/user_password_recovery_spec.rb]: This is junk whitespace.

File: app/views/devise/passwords/new.html.haml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+%h2 Forgot your password?
+= simple_form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :post }) do |f|
+  = f.error_notification
+  .form-inputs
```
Comment (elizamarcum) [inline app/views/devise/passwords/new.html.haml]: What are the .form-inputs and .form-actions divs for?

**Comment by RJFryman** [comment]
Also inadvertently made login via username or email possible.

**Comment by RJFryman** [comment]
apparently the form inputs and actions are needed or the functionality breaks?

**Pull request merged**